lib/Xud.ts

```diff
@@ -65,8 +67,34 @@ class Xud extends EventEmitter {
       this.db = new DB(loggers.db, this.config.dbpath);
       await this.db.init(this.config.network, this.config.initdb);
 
-      const nodeKey = await NodeKey.load(this.config.xudir, this.config.instanceid);
-      this.logger.info(`Local nodePubKey is ${nodeKey.nodePubKey}`);
+      this.swapClientManager = new SwapClientManager(this.config, loggers);
+      await this.swapClientManager.init(this.db.models);
+
+      const nodeKeyPath = NodeKey.getPath(this.config.xudir, this.config.instanceid);
+      let nodeKey: NodeKey;
+      if (this.config.noencrypt) {
+        nodeKey = await NodeKey.load(nodeKeyPath);
+      } else {
+        const nodeKeyExists = await fs.access(nodeKeyPath).then(() => true).catch(() => false);
+        const initService = new InitService(this.swapClientManager, nodeKeyPath, nodeKeyExists);
+
+        const initRpcServer = new GrpcServer(loggers.rpc);
+        initRpcServer.addXudInitService(initService);
+        await initRpcServer.listen(
+          this.config.rpc.port,
+          this.config.rpc.host,
+          path.join(this.config.xudir, 'tls.cert'),
+          path.join(this.config.xudir, 'tls.key'),
+        );
+
+        this.logger.info('awaiting password...');
```

Comment:
For better user experience I'd mention the `xucli unlock/create` commands here like lnd does.